src/batchUploader.ts

```diff
@@ -167,29 +167,35 @@ export class BatchUploader {
      * @param event event that should be queued
      */
     public queueEvent(event: SDKEvent): void {
-        if (!isEmpty(event)) {
-            this.eventsQueuedForProcessing.push(event);
-            if (this.offlineStorageEnabled && this.eventVault) {
-                this.eventVault.store(this.eventsQueuedForProcessing);
-            }
-            this.mpInstance.Logger.verbose(
-                `Queuing event: ${JSON.stringify(event)}`
-            );
-            this.mpInstance.Logger.verbose(
-                `Queued event count: ${this.eventsQueuedForProcessing.length}`
-            );
+        if (isEmpty(event)) {
+            return;
+        }
 
-            // TODO: Remove this check once the v2 code path is removed
-            //       https://go.mparticle.com/work/SQDSDKS-3720
-            if (
-                !this.batchingEnabled ||
-                Types.TriggerUploadType[event.EventDataType]
-            ) {
-                this.prepareAndUpload(false, false);
-            }
+        const { verbose } = this.mpInstance.Logger;
+
+        this.eventsQueuedForProcessing.push(event);
+        if (this.offlineStorageEnabled && this.eventVault) {
+            this.eventVault.store(this.eventsQueuedForProcessing);
+        }
+
+        verbose(`Queuing event: ${JSON.stringify(event)}`);
+        verbose(`Queued event count: ${this.eventsQueuedForProcessing.length}`);
+
+        if (this.shouldTriggerImmediateUpload(event.EventDataType)) {
+            this.prepareAndUpload(false, false);
         }
     }
 
+    // https://go.mparticle.com/work/SQDSDKS-3720
+    private shouldTriggerImmediateUpload (eventDataType: number): boolean {
+        const priorityEvents = [
+            MessageType.Commerce,
+            MessageType.UserIdentityChange,
+        ] as const;
+
+        return !this.batchingEnabled || priorityEvents.includes(eventDataType as typeof priorityEvents[number]);
```

Comment:
Sort of. `number` is actually being used as a `type` and not as a variable. This is saying that `eventDataType` should be treated as a type of `priorityEvent` which is an array of numbers.

This will transpile to `priorityEvents.includes(eventDataType)` but allows TypeScript to enforce that `eventDataType` is treated as a parameter that is compatible with the members of `priorityEvents`.